errors/abci.go

```diff
@@ -84,12 +74,9 @@ func abciCodespace(err error) string {
 	}
 
 	for {
-		if c, ok := err.(codespacer); ok {
-			return c.Codespace()
-		}
-
-		if c, ok := err.(causer); ok {
-			err = c.Cause()
+		var customErr *Error
+		if errors.As(err, &customErr) {
+			return customErr.Codespace()
```

Comment:
personal preference: `abciCode()` and `abciCodespace()` are called by `ABCIInfo()` only. IMHO it makes sense to inline the methods so that `customErr` is only fetched once in `ABCIInfo()`.